x-pack/plugins/ml/server/routes/data_frame_analytics.ts

```diff
@@ -671,6 +694,53 @@ export function dataFrameAnalyticsRoutes({ router, mlLicense, routeGuard }: Rout
     })
   );
 
+  /**
+   * @apiGroup DataFrameAnalytics
+   *
+   * @api {get} api/data_frame/analytics/fields/:indexPattern Get index pattern fields for analytics
+   * @apiName GetDataFrameAnalyticsIndexFields
+   * @apiDescription Retrieve the index fields for analytics
+   */
+  router.get(
+    {
+      path: '/api/ml/data_frame/analytics/fields/{indexPattern}',
+      validate: {
+        params: analyticsFieldsParamsSchema,
+        query: analyticsFieldsQuerySchema,
+      },
+      options: {
+        tags: ['access:ml:canGetJobs'],
+      },
+    },
+    routeGuard.fullLicenseAPIGuard(async ({ client, request, response, context }) => {
+      try {
+        const { indexPattern } = request.params;
+        const isRollup = request.query.rollup === 'true';
+        const savedObjectsClient = context.core.savedObjects.client;
+        const fieldService = fieldServiceProvider(
+          indexPattern,
+          isRollup,
+          client,
+          savedObjectsClient
+        );
+        const { fields, aggs } = await fieldService.getData(true);
+        convertForStringify(aggs, fields);
+        const fieldsWithoutDocCount = fields.filter(({ id }) => id !== _DOC_COUNT && id !== _TIER);
```

Comment:
Is it possible to filter using the `metadata_field` property rather than using field `id`, as was done in https://github.com/elastic/kibana/pull/96548?

Reply:
The new class gets the fields from the field service so that should already be filtered with the changes you refer to. I don't see `_tier` listed as a metadata field (https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-fields.html) so I think we still might need to keep the check against `_TIER` - though we can remove the doc count check since that is a metadata field that will be filtered out.